ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenBalanceCriteria.java

```diff
@@ -0,0 +1,93 @@
+package net.corda.v5.ledger.utxo.token.selection;
+
+import net.corda.v5.base.types.MemberX500Name;
+import net.corda.v5.ledger.utxo.observer.UtxoTokenPoolKey;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.MessageFormat;
+import java.util.Objects;
+
```

Comment:
This effectively needs to be a clone of `TokenSelectionCriteria` the only difference will be we won't have `targetAmount`